.github/workflows/preview-build.yml

```
@@ -516,6 +517,81 @@ jobs:
               environment_url: `https://docs-v3-preview.elastic.dev${process.env.LANDING_PAGE_PATH}`,
               log_url: `https://github.com/${context.repo.owner}/${context.repo.repo}/actions/runs/${context.runId}`,
             })
+
+      - name: Create Assembler Deployment
+        if: >
+          github.repository == 'elastic/docs-builder'
+          && steps.deployment.outputs.result
+          && contains(github.event.pull_request.labels.*.name, 'assembler-preview')
+        uses: actions/github-script@v8
+        id: assembler-deployment
+        env:
+          PR_NUMBER: ${{ github.event.pull_request.number }}
+          REF: ${{ github.event.pull_request.head.sha }}
+        with:
+          result-encoding: string
+          script: |
+            const { owner, repo } = context.repo;
+            const prNumber = process.env.PR_NUMBER;
+            const environment = 'assembler-preview';
+            const task = `assembler-preview-${prNumber}`;
+            const deployment = await github.rest.repos.createDeployment({
+                owner,
+                repo,
+                environment,
+                task,
+                ref: process.env.REF,
+                auto_merge: false,
+                transient_environment: true,
+                required_contexts: [],
+            })
+            await github.rest.repos.createDeploymentStatus({
+                deployment_id: deployment.data.id,
+                owner,
+                repo,
+                state: "in_progress",
+                log_url: `https://github.com/${context.repo.owner}/${context.repo.repo}/actions/runs/${context.runId}`,
+            })
+            return deployment.data.id
+
+      - name: Build assembled documentation
+        id: internal-assembler-build
+        if: steps.assembler-deployment.outputs.result
+        env:
+          ASSEMBLER_PREVIEW_PATH_PREFIX: ${{ github.repository }}/assembled-docs/${{ github.event.pull_request.number }}
+        run: |
+          yq -i ".environments.preview.path_prefix = \"${ASSEMBLER_PREVIEW_PATH_PREFIX}\"" config/assembler.yml
+          dotnet run --project src/tooling/docs-builder -- assemble --skip-private-repositories --environment preview
+
+      - name: Upload assembled docs to S3
+        id: assembler-s3-upload
+        if: steps.internal-assembler-build.outcome == 'success'
+        env:
+          AWS_RETRY_MODE: standard
+          AWS_MAX_ATTEMPTS: 6
+          ASSEMBLER_PREVIEW_PATH_PREFIX: ${{ github.repository }}/assembled-docs/${{ github.event.pull_request.number }}
```

Can we change this to something like? (just a suggestion)

```suggestion
          ASSEMBLER_PREVIEW_PATH_PREFIX: ${{ github.repository }}/pull/${{ github.event.pull_request.number }}/assembled-docs
```
My idea for this was that you can quickly guess the GitHub PR URL.

E.g.

`https://docs-v3-preview.elastic.dev/elastic/docs-builder/pull/2379`

becomes 

`https://github.com/elastic/docs-builder/pull/2379`

by replacing `docs-v3-preview.elastic.dev` with `github.com`